TortoiseGitToolbar/Config/TortoiseGitToolbar.vsct

```diff
@@ -123,35 +123,42 @@
           <ButtonText>Stash Save</ButtonText>
         </Strings>
       </Button>
-      <Button guid="guidTortoiseGitToolbarCmdSet" id="cmdSwitch" priority="0x0115" type="Button">
+      <Button guid="guidTortoiseGitToolbarCmdSet" id="cmdStashList" priority="0x0115" type="Button">
+        <Parent guid="guidTortoiseGitToolbarCmdSet" id="ToolbarGroup" />
+        <Icon guid="guidImages" id="iconStashList" />
+        <Strings>
+          <ButtonText>Stash List</ButtonText>

```

Comment:
I think what you have is fine for now. In the future, we can always run Git first to test if the stash is empty and show a dialog if needed.